chanbackup/single.go

```diff
@@ -125,12 +125,22 @@ func NewSingle(channel *channeldb.OpenChannel,
 	// go to recover, we'll present this in order to derive the private
 	// key.
 	_, shaChainPoint := btcec.PrivKeyFromBytes(btcec.S256(), b.Bytes())
+	
+	// If a channel is unconfirmed, the block height of the ShortChannelID
+	// is zero. This will lead to problems when trying to restore that
+	// channel as the spend notifier would get a height hint of zero.
+	// To work around that problem, we add the channel broadcast height
+	// to the channel ID so we can use that as height hint on restore.
+	chanId := lnwire.NewShortChanIDFromInt(channel.ShortChanID().ToUint64())
```

Comment:
```suggestion
	chanId := channel.ShortChanID()
```

Reply:
Ah, since the return value of `channel.ShortChanID()` is not a pointer, we get a clone of the value automatically?

Reply:
Exactly :)